### PR TITLE
Vagrantfile for setting up a machine with pre-configured examples

### PR DESCRIPTION
**Pull request by fchirigati**
This provides updates to the Vagrantfile and to the installation script. It also updates the README files with instructions on how to run the examples in the machine.

All the examples were tested and work.
Packing with ReproZip has not been tested yet, but ReproZip is installed in the machine.

### CODE REVIEW

**Comment by fchirigati** [comment]
@remram44 Could you try the machine? Also, feel free to make any modifications.

File: bechdel-test/bechdel.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 ## Revenue Data
 numbers_df = pd.read_csv('revenue.csv',encoding='utf8',index_col=0)
-numbers_df['Released'] = pd.to_datetime(numbers_df['Released'],unit='D')
+numbers_df['Released'] = pd.to_datetime(numbers_df['Released'])
```
Comment (remram44) [inline bechdel-test/bechdel.py]: Isn't it better to keep the original notebook's code, locking in the appropriate pandas version?

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
+export DEBIAN_FRONTEND=noninteractive
+
+echo '>> Updating information about packages...'
+sudo sh -c 'echo "deb http://cran.rstudio.com/bin/linux/ubuntu trusty/" >> /etc/apt/sources.list' # R
+gpg --keyserver keyserver.ubuntu.com --recv-key E084DAB9
+gpg -a --export E084DAB9 | sudo apt-key add -
+sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ wily-pgdg main" >> /etc/apt/sources.list.d/pgdg.list' # PostgreSQL
+wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
```
Comment (remram44) [inline install.sh]: Can't the postgresql provided by Ubuntu be used? If this is for stacked-up, it's what I always used...

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+echo '>> Installing xfce4 and VirtualBox guest tools...'
+sudo apt-get -y install xfce4 virtualbox-guest-dkms virtualbox-guest-utils virtualbox-guest-x11
+sudo sh -c 'echo "allowed_users=anybody" > /etc/X11/Xwrapper.config'
+
+echo '>> Installing pip...'
+wget --quiet https://bootstrap.pypa.io/get-pip.py
+sudo apt-get -y install python
+sudo python get-pip.py
+sudo apt-get -y install python-dev build-essential
```
Comment (remram44) [inline install.sh]: Here also, Ubuntu repositories provide pip & virtualenv...

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+echo '>> Installing dependencies for digits-sklearn-opencv...'
+mkvirtualenv --system-site-packages digits-sklearn-opencv
+cd digits-sklearn-opencv/
+pip install -r requirements.txt
+cd ../../
+sudo apt-get -y install cmake pkg-config
+sudo apt-get -y install libjpeg8-dev libtiff5-dev libjasper-dev libpng12-dev
+sudo apt-get -y install libgtk2.0-dev
+sudo apt-get -y install libatlas-base-dev gfortran
+git clone https://github.com/opencv/opencv.git
```
Comment (remram44) [inline install.sh]: Using OpenCV 2.4 (from Ubuntu) means we can use the original experiment code, don't have to compile and match the current RPZ file. Why the update? Compilation takes hours too...

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Vagrant.configure('2') do |config|
+	config.vm.box = "bento/ubuntu-15.04"
+        
+        config.vm.provider "virtualbox" do |vb|
+            vb.name = "reprozip-examples"
+            # vb.gui = true
+        end
+
+        config.vm.network "forwarded_port", guest: 8000, host: 8111
```
Comment (remram44) [inline Vagrantfile]: Why change the port number?

**Review by remram44 — CHANGES_REQUESTED** [review]
This is really long to build (and failed when I tried) and uses weird versions that require changes to our examples.

File: bechdel-test/bechdel.py (hunk 1 above)
Comment (fchirigati) [reply bechdel-test/bechdel.py]: I had already changed it back (check [here](https://github.com/ViDA-NYU/reprozip-examples/pull/6/commits/53a452231632ffa5e21fc524d9e197d06cb00372)). Do you happen to know why this is not being reflected? Perhaps when I moved the commits to a new branch I may have done something wrong?

File: Vagrantfile (hunk 5 above)
Comment (fchirigati) [reply Vagrantfile]: I wanted to be consistent with the original repository ([sdp_curricula](https://github.com/fchirigati/sdp_curricula)).

File: install.sh (hunk 3 above)
Comment (fchirigati) [reply install.sh]: pip: either way is fine for me.
virtualenv: I thought the regular way to install it was using pip (as I do).

File: install.sh (hunk 4 above)
Comment (fchirigati) [reply install.sh]: I don't know any other ways to install OpenCV -- does it come with the Ubuntu distribution? This was the standard compilation / installation step I found in many forums on the Web for Ubuntu. Let me know if there is any other way to install OpenCV.
Also, I did not have to change any of the code for OpenCV 3.0.0, but you are right that it is better to use the same one from the RPZ package (I did not even realize that), so no issues on that.

File: bechdel-test/bechdel.py (hunk 1 above)
Comment (remram44) [reply bechdel-test/bechdel.py]: There's a lot of weird merges happening here, I'm not too sure what's up (that's also why I protected the branch).

File: install.sh (hunk 2 above)
Comment (fchirigati) [reply install.sh]: I had an issue with broken dependencies (as far as I remember), and some forums suggested to use this instead. But I was also using a different box, so I can try again.

**Comment by remram44** [comment]
Rewrote the Vagrant stuff, PostgreSQL is annoying. In particular in the Ubuntu box, systemd needs to be replaced by something else (upstart) which requires a reboot ([troubleshooting#systemd](http://reprozip.readthedocs.io/en/1.0.x/troubleshooting.html#systemd)).

**Comment by fchirigati** [comment]
Any reasons for removing ReproZip from the machine? Isn't the main idea to try out the different examples with ReproZip?
Also, how come PostgreSQL 9.1 was working fine before (on my previous version and on the sdp_curricula repository)? Is this a matter of version (difference between 9.1 and 9.5)? Or perhaps they were doing something different in the original repository?

**Review by fchirigati — COMMENTED** [review]
Instructions on the README files probably need to be updated.

**Review by remram44 — APPROVED** [review]

**Comment by remram44** [comment]
After hours of investigation, I finally found the component that does the resizing: VBoxClient, started through `/etc/xdg/autostart`. I'm going to rebuild Vagrant machines for the demo and for vagrant with gui ([reprozip#191](https://github.com/ViDA-NYU/reprozip/pull/191)).

Another issue we have with the example VM is that it is too recent and therefore has PostgreSQL 9.5 while only 9.1 imports the stacked-up dump without warning (but it seems it runs fine anyway afterwards).

**Pull request merged**